nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/RestartInstanceCommand.java

```diff
@@ -134,11 +144,26 @@
     private String oldPid;
 
     private AdminCommandContext context;
-    
-    private static final long WAIT_TIME_MS = 600000; // 10 minutes
-    
+
     @Override
     public void execute(AdminCommandContext ctx) {
+        CountDownLatch commandTimeout = new CountDownLatch(1);
```

Comment:
Why conflicting methods of timeout for AdminCommands?
You've still got competing timeouts in this case - why not reconfigure the `READ_TIMEOUT` in `RemoteRestAdminCommand` to be configurable?